{"name": "Авторизация с неверными данными", "status": "broken", "statusDetails": {"message": "selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {\"method\":\"css selector\",\"selector\":\".user-icon\"}\n  (Session info: chrome=136.0.7103.114); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nStacktrace:\n\tGetHandleVerifier [0x00A3FC83+61635]\n\tGetHandleVerifier [0x00A3FCC4+61700]\n\t(No symbol) [0x008605D3]\n\t(No symbol) [0x008A899E]\n\t(No symbol) [0x008A8D3B]\n\t(No symbol) [0x008F0E12]\n\t(No symbol) [0x008CD2E4]\n\t(No symbol) [0x008EE61B]\n\t(No symbol) [0x008CD096]\n\t(No symbol) [0x0089C840]\n\t(No symbol) [0x0089D6A4]\n\tGetHandleVerifier [0x00CC45A3+2701795]\n\tGetHandleVerifier [0x00CBFD26+2683238]\n\tGetHandleVerifier [0x00CDAA6E+2793134]\n\tGetHandleVerifier [0x00A56945+155013]\n\tGetHandleVerifier [0x00A5D02D+181357]\n\tGetHandleVerifier [0x00A474D8+92440]\n\tGetHandleVerifier [0x00A47680+92864]\n\tGetHandleVerifier [0x00A32070+5296]\n\tBaseThreadInitThunk [0x7694FCC9+25]\n\tRtlGetAppContainerNamedObjectPath [0x770982AE+286]\n\tRtlGetAppContainerNamedObjectPath [0x7709827E+238]", "trace": "self = <test_ui.TestAuthUI object at 0x0000028880F31A90>, driver = <selenium.webdriver.chrome.webdriver.WebDriver (session=\"e9a3412a51afe1d3660b23959418d7a8\")>\n\n    @allure.title(\"Авторизация с неверными данными\")\n    @allure.description(\"Проверяем, что при неверном логине или пароле отображается ошибка\")\n    @allure.severity(allure.severity_level.NORMAL)\n    def test_invalid_credentials(self, driver):\n        with allure.step(\"Открываем главную страницу\"):\n            driver.get(BASE_URL)\n    \n        with allure.step(\"Открываем форму авторизации\"):\n>           driver.find_element(By.CSS_SELECTOR, \".user-icon\").click()\n\ntests\\test_ui.py:47: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n..\\..\\..\\AppData\\Local\\Programs\\Python\\Python313\\Lib\\site-packages\\selenium\\webdriver\\remote\\webdriver.py:898: in find_element\n    return self.execute(Command.FIND_ELEMENT, {\"using\": by, \"value\": value})[\"value\"]\n..\\..\\..\\AppData\\Local\\Programs\\Python\\Python313\\Lib\\site-packages\\selenium\\webdriver\\remote\\webdriver.py:429: in execute\n    self.error_handler.check_response(response)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nself = <selenium.webdriver.remote.errorhandler.ErrorHandler object at 0x0000028880F49E80>\nresponse = {'status': 404, 'value': '{\"value\":{\"error\":\"no such element\",\"message\":\"no such element: Unable to locate element: {\\...\\n\\\\tRtlGetAppContainerNamedObjectPath [0x770982AE+286]\\\\n\\\\tRtlGetAppContainerNamedObjectPath [0x7709827E+238]\\\\n\"}}'}\n\n    def check_response(self, response: Dict[str, Any]) -> None:\n        \"\"\"Checks that a JSON response from the WebDriver does not have an\n        error.\n    \n        :Args:\n         - response - The JSON response from the WebDriver server as a dictionary\n           object.\n    \n        :Raises: If the response contains an error message.\n        \"\"\"\n        status = response.get(\"status\", None)\n        if not status or status == ErrorCode.SUCCESS:\n            return\n        value = None\n        message = response.get(\"message\", \"\")\n        screen: str = response.get(\"screen\", \"\")\n        stacktrace = None\n        if isinstance(status, int):\n            value_json = response.get(\"value\", None)\n            if value_json and isinstance(value_json, str):\n                import json\n    \n                try:\n                    value = json.loads(value_json)\n                    if len(value) == 1:\n                        value = value[\"value\"]\n                    status = value.get(\"error\", None)\n                    if not status:\n                        status = value.get(\"status\", ErrorCode.UNKNOWN_ERROR)\n                        message = value.get(\"value\") or value.get(\"message\")\n                        if not isinstance(message, str):\n                            value = message\n                            message = message.get(\"message\")\n                    else:\n                        message = value.get(\"message\", None)\n                except ValueError:\n                    pass\n    \n        exception_class: Type[WebDriverException]\n        e = ErrorCode()\n        error_codes = [item for item in dir(e) if not item.startswith(\"__\")]\n        for error_code in error_codes:\n            error_info = getattr(ErrorCode, error_code)\n            if isinstance(error_info, list) and status in error_info:\n                exception_class = getattr(ExceptionMapping, error_code, WebDriverException)\n                break\n        else:\n            exception_class = WebDriverException\n    \n        if not value:\n            value = response[\"value\"]\n        if isinstance(value, str):\n            raise exception_class(value)\n        if message == \"\" and \"message\" in value:\n            message = value[\"message\"]\n    \n        screen = None  # type: ignore[assignment]\n        if \"screen\" in value:\n            screen = value[\"screen\"]\n    \n        stacktrace = None\n        st_value = value.get(\"stackTrace\") or value.get(\"stacktrace\")\n        if st_value:\n            if isinstance(st_value, str):\n                stacktrace = st_value.split(\"\\n\")\n            else:\n                stacktrace = []\n                try:\n                    for frame in st_value:\n                        line = frame.get(\"lineNumber\", \"\")\n                        file = frame.get(\"fileName\", \"<anonymous>\")\n                        if line:\n                            file = f\"{file}:{line}\"\n                        meth = frame.get(\"methodName\", \"<anonymous>\")\n                        if \"className\" in frame:\n                            meth = f\"{frame['className']}.{meth}\"\n                        msg = \"    at %s (%s)\"\n                        msg = msg % (meth, file)\n                        stacktrace.append(msg)\n                except TypeError:\n                    pass\n        if exception_class == UnexpectedAlertPresentException:\n            alert_text = None\n            if \"data\" in value:\n                alert_text = value[\"data\"].get(\"text\")\n            elif \"alert\" in value:\n                alert_text = value[\"alert\"].get(\"text\")\n            raise exception_class(message, screen, stacktrace, alert_text)  # type: ignore[call-arg]  # mypy is not smart enough here\n>       raise exception_class(message, screen, stacktrace)\nE       selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {\"method\":\"css selector\",\"selector\":\".user-icon\"}\nE         (Session info: chrome=136.0.7103.114); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nE       Stacktrace:\nE       \tGetHandleVerifier [0x00A3FC83+61635]\nE       \tGetHandleVerifier [0x00A3FCC4+61700]\nE       \t(No symbol) [0x008605D3]\nE       \t(No symbol) [0x008A899E]\nE       \t(No symbol) [0x008A8D3B]\nE       \t(No symbol) [0x008F0E12]\nE       \t(No symbol) [0x008CD2E4]\nE       \t(No symbol) [0x008EE61B]\nE       \t(No symbol) [0x008CD096]\nE       \t(No symbol) [0x0089C840]\nE       \t(No symbol) [0x0089D6A4]\nE       \tGetHandleVerifier [0x00CC45A3+2701795]\nE       \tGetHandleVerifier [0x00CBFD26+2683238]\nE       \tGetHandleVerifier [0x00CDAA6E+2793134]\nE       \tGetHandleVerifier [0x00A56945+155013]\nE       \tGetHandleVerifier [0x00A5D02D+181357]\nE       \tGetHandleVerifier [0x00A474D8+92440]\nE       \tGetHandleVerifier [0x00A47680+92864]\nE       \tGetHandleVerifier [0x00A32070+5296]\nE       \tBaseThreadInitThunk [0x7694FCC9+25]\nE       \tRtlGetAppContainerNamedObjectPath [0x770982AE+286]\nE       \tRtlGetAppContainerNamedObjectPath [0x7709827E+238]\n\n..\\..\\..\\AppData\\Local\\Programs\\Python\\Python313\\Lib\\site-packages\\selenium\\webdriver\\remote\\errorhandler.py:232: NoSuchElementException"}, "description": "Проверяем, что при неверном логине или пароле отображается ошибка", "steps": [{"name": "Открываем главную страницу", "status": "passed", "start": 1747575436301, "stop": 1747575478730}, {"name": "Открываем форму авторизации", "status": "broken", "statusDetails": {"message": "selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {\"method\":\"css selector\",\"selector\":\".user-icon\"}\n  (Session info: chrome=136.0.7103.114); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nStacktrace:\n\tGetHandleVerifier [0x00A3FC83+61635]\n\tGetHandleVerifier [0x00A3FCC4+61700]\n\t(No symbol) [0x008605D3]\n\t(No symbol) [0x008A899E]\n\t(No symbol) [0x008A8D3B]\n\t(No symbol) [0x008F0E12]\n\t(No symbol) [0x008CD2E4]\n\t(No symbol) [0x008EE61B]\n\t(No symbol) [0x008CD096]\n\t(No symbol) [0x0089C840]\n\t(No symbol) [0x0089D6A4]\n\tGetHandleVerifier [0x00CC45A3+2701795]\n\tGetHandleVerifier [0x00CBFD26+2683238]\n\tGetHandleVerifier [0x00CDAA6E+2793134]\n\tGetHandleVerifier [0x00A56945+155013]\n\tGetHandleVerifier [0x00A5D02D+181357]\n\tGetHandleVerifier [0x00A474D8+92440]\n\tGetHandleVerifier [0x00A47680+92864]\n\tGetHandleVerifier [0x00A32070+5296]\n\tBaseThreadInitThunk [0x7694FCC9+25]\n\tRtlGetAppContainerNamedObjectPath [0x770982AE+286]\n\tRtlGetAppContainerNamedObjectPath [0x7709827E+238]\n\n", "trace": "  File \"C:\\Users\\admin\\Desktop\\Фиальный проект\\final-project\\tests\\test_ui.py\", line 47, in test_invalid_credentials\n    driver.find_element(By.CSS_SELECTOR, \".user-icon\").click()\n    ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\admin\\AppData\\Local\\Programs\\Python\\Python313\\Lib\\site-packages\\selenium\\webdriver\\remote\\webdriver.py\", line 898, in find_element\n    return self.execute(Command.FIND_ELEMENT, {\"using\": by, \"value\": value})[\"value\"]\n           ~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\admin\\AppData\\Local\\Programs\\Python\\Python313\\Lib\\site-packages\\selenium\\webdriver\\remote\\webdriver.py\", line 429, in execute\n    self.error_handler.check_response(response)\n    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^\n  File \"C:\\Users\\admin\\AppData\\Local\\Programs\\Python\\Python313\\Lib\\site-packages\\selenium\\webdriver\\remote\\errorhandler.py\", line 232, in check_response\n    raise exception_class(message, screen, stacktrace)\n"}, "start": 1747575478730, "stop": 1747575478822}], "start": 1747575436301, "stop": 1747575478823, "uuid": "ae7d74e3-c008-4881-8b27-ced6b1bc51ab", "historyId": "bf50f403c9d14545bd5b6569fa8c38a8", "testCaseId": "bf50f403c9d14545bd5b6569fa8c38a8", "fullName": "tests.test_ui.TestAuthUI#test_invalid_credentials", "labels": [{"name": "feature", "value": "Авторизация"}, {"name": "severity", "value": "normal"}, {"name": "tag", "value": "ui"}, {"name": "parentSuite", "value": "tests"}, {"name": "suite", "value": "test_ui"}, {"name": "subSuite", "value": "TestAuthUI"}, {"name": "host", "value": "DESKTOP-4F3BQLH"}, {"name": "thread", "value": "3096-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "tests.test_ui"}]}